codetoanalyze/cpp/pulse/aliasing.cpp, call_ifthenderef_true_null_bad, 0, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,when calling `ifthenderef` here,invalid access occurs here]
codetoanalyze/cpp/pulse/aliasing.cpp, call_ifnotthenderef_false_null_bad, 0, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,when calling `ifnotthenderef` here,invalid access occurs here]
codetoanalyze/cpp/pulse/aliasing.cpp, null_test_after_deref_latent_FN, 1, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/basic_string.cpp, use_range_of_invalidated_temporary_string_bad, 2, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,when calling `setLanguage` here,assigned,in call to `std::basic_string::~basic_string()` (modelled),was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `setLanguage`,assigned,in call to `Range::Range`,in call to `std::basic_string::data()` (modelled),assigned,return from call to `Range::Range`,return from call to `setLanguage`,when calling `Range::operator[]` here,invalid access occurs here]
codetoanalyze/cpp/pulse/basics.cpp, multiple_invalidations_branch_latent, 6, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/basics.cpp, multiple_invalidations_branch_latent, 6, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/basics.cpp, multiple_invalidations_branch_latent, 6, USE_AFTER_DELETE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/basics.cpp, multiple_invalidations_branch_latent, 6, USE_AFTER_DELETE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/basics.cpp, multiple_invalidations_loop_latent, 5, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/basics.cpp, multiple_invalidations_loop_latent, 5, USE_AFTER_DELETE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/basics.cpp, multiple_invalidations_loop_latent, 8, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,assigned,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/basics.cpp, multiple_invalidations_loop_latent, 8, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/basics.cpp, multiple_invalidations_loop_latent, 8, USE_AFTER_DELETE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, ref_capture_destroy_invoke_bad, 6, USE_AFTER_LIFETIME, no_bucket, ERROR, [invalidation part of the trace starts here,variable `s` declared here,is the address of a stack variable `s` whose lifetime has ended,use-after-lifetime part of the trace starts here,variable `s` declared here,value captured by ref as `s`,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, implicit_ref_capture_destroy_invoke_bad, 6, USE_AFTER_LIFETIME, no_bucket, ERROR, [invalidation part of the trace starts here,variable `s` declared here,is the address of a stack variable `s` whose lifetime has ended,use-after-lifetime part of the trace starts here,variable `s` declared here,value captured by ref as `s`,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, reassign_lambda_capture_destroy_invoke_bad, 9, USE_AFTER_LIFETIME, no_bucket, ERROR, [invalidation part of the trace starts here,variable `s` declared here,is the address of a stack variable `s` whose lifetime has ended,use-after-lifetime part of the trace starts here,variable `s` declared here,value captured by ref as `s`,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, ref_capture_return_local_lambda_bad, 7, STACK_VARIABLE_ADDRESS_ESCAPE, no_bucket, ERROR, [variable `x` declared here,value captured by ref as `x`,in call to `ref_capture_return_local_lambda_bad::lambda_closures.cpp:129:12::operator()`,returned,return from call to `ref_capture_return_local_lambda_bad::lambda_closures.cpp:129:12::operator()`,returned here]
codetoanalyze/cpp/pulse/closures.cpp, call_lambda_bad, 4, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `call_lambda_bad::lambda_closures.cpp:164:12::operator()` here,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, call_lambda_std_fun_bad, 5, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `call_lambda_std_fun_bad::lambda_closures.cpp:172:7::operator()` here,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, call_std_fun_constructor_bad, 2, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/closures.cpp, call_std_fun_constructor_bad, 5, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `call_std_fun_constructor_bad::lambda_closures.cpp:179:32::operator()` here,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, capture_by_value_bad, 7, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, capture_by_ref_bad, 7, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, capture_by_value_init_bad, 7, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, capture_by_ref_init_bad, 7, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, ref_capture_by_value_bad, 8, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, ref_capture_by_ref_bad, 8, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, struct_capture_by_ref_bad, 7, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, struct_capture_by_val_bad, 7, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, struct_capture_by_val_ok_FP, 7, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, update_inside_lambda_as_argument_ok_FP, 1, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `update_inside_lambda_as_argument`,is assigned to the null pointer,assigned,returned,return from call to `update_inside_lambda_as_argument`,invalid access occurs here]
codetoanalyze/cpp/pulse/conditional_temporaries.cpp, condtemp::X::__infer_inner_destructor_~X, 1, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,when calling `condtemp::X::name` here,invalid access occurs here]
codetoanalyze/cpp/pulse/conditional_temporaries.cpp, condtemp::FP_track_copy_operations_one_copy_ok, 3, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/conditional_temporaries.cpp, condtemp::FP_track_copy_operations_one_copy_ok, 17, NULLPTR_DEREFERENCE, no_bucket, ERROR, [condition is false,in call to `condtemp::X::X`,in call to `condtemp::X::copy_from`,taking "then" branch,assigned,is assigned to the constant 1,assigned,return from call to `condtemp::X::copy_from`,return from call to `condtemp::X::X`,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/conditional_temporaries.cpp, condtemp::FP_track_copy_operations_complex_ok, 16, NULLPTR_DEREFERENCE, no_bucket, ERROR, [condition is true,in call to `condtemp::copy`,in call to `condtemp::X::X`,in call to `condtemp::X::copy_from`,taking "then" branch,assigned,is assigned to the constant 1,assigned,return from call to `condtemp::X::copy_from`,return from call to `condtemp::X::X`,return from call to `condtemp::copy`,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/conditionals.cpp, add_test3_latent, 3, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/conditionals.cpp, add_test3_latent, 3, USE_AFTER_FREE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/conditionals.cpp, add_test5_latent, 5, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/conditionals.cpp, add_test5_latent, 5, USE_AFTER_FREE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/const_volatile_type.cpp, test_volatile1_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::unique_ptr<int,std::default_delete<int>>::unique_ptr<true,_void>`,is assigned to the null pointer,assigned,in call to `std::__compressed_pair<int*,std::default_delete<int>>::__compressed_pair<1d1b5522baf3d24e>`,in call to `std::forward<volatile_int_*>`,returned,return from call to `std::forward<volatile_int_*>`,in call to `std::__compressed_pair_elem<int*,0,0>::__compressed_pair_elem<volatile_int_*,_void>`,in call to `std::forward<volatile_int_*>`,returned,return from call to `std::forward<volatile_int_*>`,assigned,return from call to `std::__compressed_pair_elem<int*,0,0>::__compressed_pair_elem<volatile_int_*,_void>`,return from call to `std::__compressed_pair<int*,std::default_delete<int>>::__compressed_pair<1d1b5522baf3d24e>`,return from call to `std::unique_ptr<int,std::default_delete<int>>::unique_ptr<true,_void>`,in call to `std::unique_ptr<int,std::default_delete<int>>::unique_ptr`,in call to `std::unique_ptr<int,std::default_delete<int>>::release`,in call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,in call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,assigned,returned,return from call to `std::unique_ptr<int,std::default_delete<int>>::release`,assigned,in call to `std::__compressed_pair<int*,std::default_delete<int>>::__compressed_pair<5cda69cf4fd276d7>`,in call to `std::forward<volatile_int_*>`,returned,return from call to `std::forward<volatile_int_*>`,in call to `std::__compressed_pair_elem<int*,0,0>::__compressed_pair_elem<volatile_int_*,_void>`,in call to `std::forward<volatile_int_*>`,returned,return from call to `std::forward<volatile_int_*>`,assigned,return from call to `std::__compressed_pair_elem<int*,0,0>::__compressed_pair_elem<volatile_int_*,_void>`,return from call to `std::__compressed_pair<int*,std::default_delete<int>>::__compressed_pair<5cda69cf4fd276d7>`,return from call to `std::unique_ptr<int,std::default_delete<int>>::unique_ptr`,in call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,in call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,in call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,returned,return from call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/const_volatile_type.cpp, test_volatile2_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::unique_ptr<int,std::default_delete<int>>::unique_ptr<true,_void>`,is assigned to the null pointer,assigned,in call to `std::__compressed_pair<int*,std::default_delete<int>>::__compressed_pair<1d1b5522baf3d24e>`,in call to `std::forward<volatile_int_*>`,returned,return from call to `std::forward<volatile_int_*>`,in call to `std::__compressed_pair_elem<int*,0,0>::__compressed_pair_elem<volatile_int_*,_void>`,in call to `std::forward<volatile_int_*>`,returned,return from call to `std::forward<volatile_int_*>`,assigned,return from call to `std::__compressed_pair_elem<int*,0,0>::__compressed_pair_elem<volatile_int_*,_void>`,return from call to `std::__compressed_pair<int*,std::default_delete<int>>::__compressed_pair<1d1b5522baf3d24e>`,return from call to `std::unique_ptr<int,std::default_delete<int>>::unique_ptr<true,_void>`,in call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,in call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,in call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,returned,return from call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/const_volatile_type.cpp, test_volatile3_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::shared_ptr<int>::shared_ptr`,is assigned to the null pointer,assigned,return from call to `std::shared_ptr<int>::shared_ptr`,in call to `std::shared_ptr<int>::shared_ptr`,assigned,return from call to `std::shared_ptr<int>::shared_ptr`,in call to `std::shared_ptr<int>::operator*`,returned,return from call to `std::shared_ptr<int>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/const_volatile_type.cpp, test_volatile4_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::shared_ptr<int>::shared_ptr`,is assigned to the null pointer,assigned,return from call to `std::shared_ptr<int>::shared_ptr`,in call to `std::shared_ptr<int>::operator*`,returned,return from call to `std::shared_ptr<int>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/const_volatile_type.cpp, test_const1_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::unique_ptr<int const ,std::default_delete<int const >>::unique_ptr<true,_void>`,is assigned to the null pointer,assigned,in call to `std::__compressed_pair<int const *,std::default_delete<int const >>::__compressed_pair<const_int_*,_std::__default_init_tag>`,in call to `std::forward<const_int_*>`,returned,return from call to `std::forward<const_int_*>`,in call to `std::__compressed_pair_elem<int const *,0,0>::__compressed_pair_elem<const_int_*,_void>`,in call to `std::forward<const_int_*>`,returned,return from call to `std::forward<const_int_*>`,assigned,return from call to `std::__compressed_pair_elem<int const *,0,0>::__compressed_pair_elem<const_int_*,_void>`,return from call to `std::__compressed_pair<int const *,std::default_delete<int const >>::__compressed_pair<const_int_*,_std::__default_init_tag>`,return from call to `std::unique_ptr<int const ,std::default_delete<int const >>::unique_ptr<true,_void>`,in call to `std::unique_ptr<int const ,std::default_delete<int const >>::unique_ptr`,in call to `std::unique_ptr<int const ,std::default_delete<int const >>::release`,in call to `std::__compressed_pair<int const *,std::default_delete<int const >>::first`,in call to `std::__compressed_pair_elem<int const *,0,0>::__get`,returned,return from call to `std::__compressed_pair_elem<int const *,0,0>::__get`,returned,return from call to `std::__compressed_pair<int const *,std::default_delete<int const >>::first`,assigned,returned,return from call to `std::unique_ptr<int const ,std::default_delete<int const >>::release`,assigned,in call to `std::__compressed_pair<int const *,std::default_delete<int const >>::__compressed_pair<4454c51fce2393b>`,in call to `std::forward<const_int_*>`,returned,return from call to `std::forward<const_int_*>`,in call to `std::__compressed_pair_elem<int const *,0,0>::__compressed_pair_elem<const_int_*,_void>`,in call to `std::forward<const_int_*>`,returned,return from call to `std::forward<const_int_*>`,assigned,return from call to `std::__compressed_pair_elem<int const *,0,0>::__compressed_pair_elem<const_int_*,_void>`,return from call to `std::__compressed_pair<int const *,std::default_delete<int const >>::__compressed_pair<4454c51fce2393b>`,return from call to `std::unique_ptr<int const ,std::default_delete<int const >>::unique_ptr`,in call to `std::unique_ptr<int const ,std::default_delete<int const >>::operator*`,in call to `std::__compressed_pair<int const *,std::default_delete<int const >>::first`,in call to `std::__compressed_pair_elem<int const *,0,0>::__get`,returned,return from call to `std::__compressed_pair_elem<int const *,0,0>::__get`,returned,return from call to `std::__compressed_pair<int const *,std::default_delete<int const >>::first`,returned,return from call to `std::unique_ptr<int const ,std::default_delete<int const >>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/const_volatile_type.cpp, test_const2_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::unique_ptr<int const ,std::default_delete<int const >>::unique_ptr<true,_void>`,is assigned to the null pointer,assigned,in call to `std::__compressed_pair<int const *,std::default_delete<int const >>::__compressed_pair<const_int_*,_std::__default_init_tag>`,in call to `std::forward<const_int_*>`,returned,return from call to `std::forward<const_int_*>`,in call to `std::__compressed_pair_elem<int const *,0,0>::__compressed_pair_elem<const_int_*,_void>`,in call to `std::forward<const_int_*>`,returned,return from call to `std::forward<const_int_*>`,assigned,return from call to `std::__compressed_pair_elem<int const *,0,0>::__compressed_pair_elem<const_int_*,_void>`,return from call to `std::__compressed_pair<int const *,std::default_delete<int const >>::__compressed_pair<const_int_*,_std::__default_init_tag>`,return from call to `std::unique_ptr<int const ,std::default_delete<int const >>::unique_ptr<true,_void>`,in call to `std::unique_ptr<int const ,std::default_delete<int const >>::operator*`,in call to `std::__compressed_pair<int const *,std::default_delete<int const >>::first`,in call to `std::__compressed_pair_elem<int const *,0,0>::__get`,returned,return from call to `std::__compressed_pair_elem<int const *,0,0>::__get`,returned,return from call to `std::__compressed_pair<int const *,std::default_delete<int const >>::first`,returned,return from call to `std::unique_ptr<int const ,std::default_delete<int const >>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/const_volatile_type.cpp, test_const3_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::shared_ptr<int const >::shared_ptr`,is assigned to the null pointer,assigned,return from call to `std::shared_ptr<int const >::shared_ptr`,in call to `std::shared_ptr<int const >::shared_ptr`,assigned,return from call to `std::shared_ptr<int const >::shared_ptr`,in call to `std::shared_ptr<int const >::operator*`,returned,return from call to `std::shared_ptr<int const >::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/const_volatile_type.cpp, test_const4_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::shared_ptr<int const >::shared_ptr`,is assigned to the null pointer,assigned,return from call to `std::shared_ptr<int const >::shared_ptr`,in call to `std::shared_ptr<int const >::operator*`,returned,return from call to `std::shared_ptr<int const >::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/deduplication.cpp, deduplication::SomeTemplatedClass<int>::lifetime_error_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,when calling `deduplication::SomeTemplatedClass<int>::templated_wrapper_delete_ok` here,is assigned to the null pointer,null pointer dereference part of the trace starts here,when calling `deduplication::SomeTemplatedClass<int>::templated_wrapper_access_ok` here,invalid access occurs here]
codetoanalyze/cpp/pulse/deduplication.cpp, deduplication::SomeTemplatedClass<int*>::lifetime_error_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,when calling `deduplication::SomeTemplatedClass<int*>::templated_wrapper_delete_ok` here,is assigned to the null pointer,null pointer dereference part of the trace starts here,when calling `deduplication::SomeTemplatedClass<int*>::templated_wrapper_access_ok` here,invalid access occurs here]
codetoanalyze/cpp/pulse/deduplication.cpp, deduplication::SomeTemplatedClass<int>::lifetime_error_bad, 2, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,when calling `deduplication::SomeTemplatedClass<int>::templated_wrapper_delete_ok` here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,when calling `deduplication::SomeTemplatedClass<int>::templated_wrapper_access_ok` here,invalid access occurs here]
codetoanalyze/cpp/pulse/deduplication.cpp, deduplication::SomeTemplatedClass<int*>::lifetime_error_bad, 2, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,when calling `deduplication::SomeTemplatedClass<int*>::templated_wrapper_delete_ok` here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,when calling `deduplication::SomeTemplatedClass<int*>::templated_wrapper_access_ok` here,invalid access occurs here]
codetoanalyze/cpp/pulse/deduplication.cpp, deduplication::templated_function_bad<_Bool>, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `deduplication::templated_delete_function<_Bool>` here,is assigned to the null pointer,null pointer dereference part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `deduplication::templated_access_function<_Bool>` here,invalid access occurs here]
codetoanalyze/cpp/pulse/deduplication.cpp, deduplication::templated_function_bad<int>, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `deduplication::templated_delete_function<int>` here,is assigned to the null pointer,null pointer dereference part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `deduplication::templated_access_function<int>` here,invalid access occurs here]
codetoanalyze/cpp/pulse/deduplication.cpp, deduplication::templated_function_bad<_Bool>, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `deduplication::templated_delete_function<_Bool>` here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `deduplication::templated_access_function<_Bool>` here,invalid access occurs here]
codetoanalyze/cpp/pulse/deduplication.cpp, deduplication::templated_function_bad<int>, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `deduplication::templated_delete_function<int>` here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `deduplication::templated_access_function<int>` here,invalid access occurs here]
codetoanalyze/cpp/pulse/deref_after_move_example.cpp, deref_after_mode_example::deref_after_move_bad, 4, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `deref_after_mode_example::Person::move_age`,in call to `std::unique_ptr<int,std::default_delete<int>>::unique_ptr`,in call to `std::unique_ptr<int,std::default_delete<int>>::release`,is assigned to the null pointer,assigned,return from call to `std::unique_ptr<int,std::default_delete<int>>::release`,return from call to `std::unique_ptr<int,std::default_delete<int>>::unique_ptr`,return from call to `deref_after_mode_example::Person::move_age`,when calling `deref_after_mode_example::Person::access_age` here,in call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,in call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,in call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,returned,return from call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/destructor.cpp, destructor::destruct_unique_deletes_pointer_bad, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,in call to `destructor::UniquePtr<destructor::X>::UniquePtr`,assigned,return from call to `destructor::UniquePtr<destructor::X>::UniquePtr`,when calling `destructor::UniquePtr<destructor::X>::~UniquePtr` here,when calling `destructor::UniquePtr<destructor::X>::__infer_inner_destructor_~UniquePtr` here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/destructor.cpp, destructor::destruct_unique_ref_deletes_pointer_bad, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,in call to `destructor::UniquePtr<destructor::X>::UniquePtr`,assigned,return from call to `destructor::UniquePtr<destructor::X>::UniquePtr`,when calling `destructor::UniquePtr<destructor::X>::~UniquePtr` here,when calling `destructor::UniquePtr<destructor::X>::__infer_inner_destructor_~UniquePtr` here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/destructor.cpp, destructor::destruct_unique_ptr_deletes_pointer_bad, 5, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,in call to `destructor::UniquePtr<destructor::X>::UniquePtr`,assigned,return from call to `destructor::UniquePtr<destructor::X>::UniquePtr`,when calling `destructor::UniquePtr<destructor::X>::~UniquePtr` here,when calling `destructor::UniquePtr<destructor::X>::__infer_inner_destructor_~UniquePtr` here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/exit_test.cpp, store_exit_null_bad, 0, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,when calling `store_exit` here,invalid access occurs here]
codetoanalyze/cpp/pulse/folly_DestructorGuard.cpp, UsingDelayedDestruction::double_delete_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/folly_DestructorGuard.cpp, UsingDelayedDestruction::double_delete_bad, 2, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/folly_DestructorGuard.cpp, UsingDelayedDestruction::latent_double_delete_ok, 1, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/frontend.cpp, frontend::deref_null_namespace_alias_ptr_bad, 4, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,when calling `frontend::some::thing::bad_ptr` here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `frontend::some::thing::bad_ptr`,allocated by call to `new` (modelled),assigned,returned,return from call to `frontend::some::thing::bad_ptr`,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/frontend.cpp, frontend::conditional_expression_bad, 5, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,condition is false,is assigned to the null pointer,assigned,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/frontend.cpp, frontend::call_Frontend_constructor_bad, 4, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `frontend::Frontend::Frontend`,parameter `n` of frontend::Frontend::Frontend,condition is false,is assigned to the constant 1,assigned,return from call to `frontend::Frontend::Frontend`,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/frontend.cpp, frontend::call_Frontend_constructor2_bad, 5, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `frontend::Frontend::Frontend`,parameter `n` of frontend::Frontend::Frontend,condition is false,is assigned to the constant 1,assigned,assigned,return from call to `frontend::Frontend::Frontend`,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/frontend.cpp, frontend::call_set_field_via_local_bad, 5, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `frontend::Frontend::set_field_via_local`,parameter `n` of frontend::Frontend::set_field_via_local,in call to `malloc (null case)` (modelled),is assigned to the null pointer,assigned,taking "else" branch,condition is false,is assigned to the constant 1,assigned,return from call to `frontend::Frontend::set_field_via_local`,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/frontend.cpp, frontend::not_boolean_bad, 8, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,allocated by call to `malloc` (modelled),assigned,taking "then" branch,condition is false,is assigned to the constant 1,assigned,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/frontend.cpp, frontend::FP_init_single_field_struct_ok, 4, NULLPTR_DEREFERENCE, no_bucket, ERROR, [variable `y` declared here,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/interprocedural.cpp, delete_then_read_bad, 2, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,when calling `wraps_read` here,when calling `wraps_read_inner` here,invalid access occurs here]
codetoanalyze/cpp/pulse/interprocedural.cpp, delete_aliased_then_read_bad, 4, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,assigned,when calling `wraps_read` here,when calling `wraps_read_inner` here,invalid access occurs here]
codetoanalyze/cpp/pulse/interprocedural.cpp, delete_then_write_bad, 2, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,when calling `wraps_delete` here,when calling `wraps_delete_inner` here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,when calling `wraps_read` here,when calling `wraps_read_inner` here,invalid access occurs here]
codetoanalyze/cpp/pulse/interprocedural.cpp, delete_inner_then_write_bad, 2, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,when calling `wraps_delete_inner` here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,when calling `wraps_read` here,when calling `wraps_read_inner` here,invalid access occurs here]
codetoanalyze/cpp/pulse/interprocedural.cpp, latent_read_write_then_delete_ok, 1, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,when calling `wraps_write` here,when calling `wraps_write_inner` here,invalid access occurs here]
codetoanalyze/cpp/pulse/interprocedural.cpp, feed_invalid_into_access_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,when calling `may_return_invalid_ptr_ok` here,allocated by call to `new` (modelled),assigned,is assigned to the null pointer,null pointer dereference part of the trace starts here,in call to `may_return_invalid_ptr_ok`,allocated by call to `new` (modelled),assigned,returned,return from call to `may_return_invalid_ptr_ok`,assigned,when calling `call_store` here,when calling `store` here,invalid access occurs here]
codetoanalyze/cpp/pulse/interprocedural.cpp, feed_invalid_into_access_bad, 2, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,when calling `may_return_invalid_ptr_ok` here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `may_return_invalid_ptr_ok`,allocated by call to `new` (modelled),assigned,returned,return from call to `may_return_invalid_ptr_ok`,assigned,when calling `call_store` here,when calling `store` here,invalid access occurs here]
codetoanalyze/cpp/pulse/interprocedural.cpp, set_x_then_crash_latent, 2, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,assigned,is assigned to the null pointer,null pointer dereference part of the trace starts here,when calling `set_first_non_null_ok` here,invalid access occurs here]
codetoanalyze/cpp/pulse/interprocedural.cpp, set_x_then_crash_latent, 4, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/join.cpp, invalidate_node_alias_latent, 12, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,assigned,is assigned to the null pointer,null pointer dereference part of the trace starts here,parameter `cond` of invalidate_node_alias_latent,assigned,taking "then" branch,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/join.cpp, invalidate_node_alias_latent, 12, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,parameter `cond` of invalidate_node_alias_latent,assigned,taking "then" branch,assigned,is assigned to the null pointer,null pointer dereference part of the trace starts here,parameter `cond` of invalidate_node_alias_latent,assigned,taking "then" branch,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/join.cpp, invalidate_node_alias_latent, 12, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,parameter `cond` of invalidate_node_alias_latent,assigned,taking "else" branch,assigned,assigned,is assigned to the null pointer,null pointer dereference part of the trace starts here,parameter `cond` of invalidate_node_alias_latent,assigned,taking "else" branch,assigned,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/join.cpp, invalidate_node_alias_latent, 12, USE_AFTER_DELETE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `cond` of invalidate_node_alias_latent,assigned,taking "then" branch,assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,parameter `cond` of invalidate_node_alias_latent,assigned,taking "then" branch,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/join.cpp, invalidate_node_alias_latent, 12, USE_AFTER_DELETE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `cond` of invalidate_node_alias_latent,assigned,taking "else" branch,assigned,assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,parameter `cond` of invalidate_node_alias_latent,assigned,taking "else" branch,assigned,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/join.cpp, invalidate_node_alias_bad, 1, NULLPTR_DEREFERENCE, no_bucket, ERROR, [*** SUPPRESSED ***,calling context starts here,in call to `invalidate_node_alias_latent`,source of the null value part of the trace starts here,parameter `cond` of invalidate_node_alias_latent,assigned,taking "then" branch,assigned,is assigned to the null pointer,null pointer dereference part of the trace starts here,parameter `cond` of invalidate_node_alias_latent,assigned,taking "then" branch,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/join.cpp, invalidate_node_alias_bad, 1, USE_AFTER_DELETE, no_bucket, ERROR, [calling context starts here,in call to `invalidate_node_alias_latent`,invalidation part of the trace starts here,parameter `cond` of invalidate_node_alias_latent,assigned,taking "then" branch,assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,parameter `cond` of invalidate_node_alias_latent,assigned,taking "then" branch,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/leaks.cpp, leaks::simple_new_bad, 0, MEMORY_LEAK_CPP, no_bucket, ERROR, [allocation part of the trace starts here,allocated by `new` here,memory becomes unreachable here]
codetoanalyze/cpp/pulse/leaks.cpp, leaks::simple_new_array_bad, 0, MEMORY_LEAK_CPP, no_bucket, ERROR, [allocation part of the trace starts here,allocated by `new[]` here,memory becomes unreachable here]
codetoanalyze/cpp/pulse/leaks.cpp, leaks::unique_ptr_pointer_bad, 1, MEMORY_LEAK_CPP, no_bucket, ERROR, [allocation part of the trace starts here,allocated by `new` here,memory becomes unreachable here]
codetoanalyze/cpp/pulse/leaks.cpp, leaks::FP_call_static_local_alloc_twice_ok, -6, MEMORY_LEAK_CPP, no_bucket, ERROR, [allocation part of the trace starts here,when calling `leaks::static_local_alloc_ok` here,allocated by `new` here,memory becomes unreachable here]
codetoanalyze/cpp/pulse/leaks.cpp, leaks::unknown_wrapper_alloc_then_leak_bad, 3, MEMORY_LEAK_CPP, no_bucket, ERROR, [allocation part of the trace starts here,allocated by `new` here,memory becomes unreachable here]
codetoanalyze/cpp/pulse/nullptr.cpp, deref_nullptr_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/nullptr.cpp, no_check_return_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `may_return_nullptr`,in call to function `choice` with no summary,taking "then" branch,is assigned to the null pointer,returned,return from call to `may_return_nullptr`,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/nullptr.cpp, std_true_type_deref_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/nullptr.cpp, std_false_type_deref_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/nullptr.cpp, call_test_after_dereference_bad, 1, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,when calling `FN_test_after_dereference_latent` here,invalid access occurs here]
codetoanalyze/cpp/pulse/nullptr.cpp, test_after_dereference2_latent, 1, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/nullptr.cpp, call_test_after_dereference2_bad, 1, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,when calling `test_after_dereference2_latent` here,invalid access occurs here]
codetoanalyze/cpp/pulse/nullptr.cpp, SomeDerivedClass::SomeDerivedClass, 0, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,when calling `SomeClass::SomeClass` here,assigned,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/nullptr.cpp, createSomeDerivedClass_from_unknown_function_ok, 1, NULLPTR_DEREFERENCE, no_bucket, ERROR, [*** SUPPRESSED ***,calling context starts here,in call to `SomeDerivedClass::SomeDerivedClass`,source of the null value part of the trace starts here,when calling `SomeClass::SomeClass` here,assigned,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/nullptr.cpp, test_call_nullptr_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/optional.cpp, not_none_check_value_ok_FP, 5, OPTIONAL_EMPTY_ACCESS, no_bucket, ERROR, [in call to `folly::Optional::Optional(Value arg)` (modelled),assigned,taking "then" branch,in call to `folly::Optional::Optional(=None)` (modelled),is assigned an empty value,invalid access occurs here]
codetoanalyze/cpp/pulse/optional.cpp, none_no_check_bad, 2, OPTIONAL_EMPTY_ACCESS, no_bucket, ERROR, [in call to `folly::Optional::Optional(=None)` (modelled),is assigned an empty value,invalid access occurs here]
codetoanalyze/cpp/pulse/optional.cpp, none_copy_ok, 2, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/optional.cpp, none_copy_bad, 2, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/optional.cpp, none_copy_bad, 3, OPTIONAL_EMPTY_ACCESS, no_bucket, ERROR, [in call to `folly::Optional::Optional(=None)` (modelled),is assigned an empty value,in call to `folly::Optional::Optional(folly::Optional<Value> arg)` (modelled),invalid access occurs here]
codetoanalyze/cpp/pulse/optional.cpp, assign_bad, 5, OPTIONAL_EMPTY_ACCESS, no_bucket, ERROR, [in call to `folly::Optional::Optional(=None)` (modelled),is assigned an empty value,in call to `folly::Optional<int>::operator=`,in call to `folly::Optional::assign(folly::Optional<Value> arg)` (modelled),return from call to `folly::Optional<int>::operator=`,invalid access occurs here]
codetoanalyze/cpp/pulse/optional.cpp, assign2_bad, 4, OPTIONAL_EMPTY_ACCESS, no_bucket, ERROR, [in call to `folly::Optional<int>::operator=`,in call to `folly::Optional::reset()` (modelled),is assigned an empty value,return from call to `folly::Optional<int>::operator=`,invalid access occurs here]
codetoanalyze/cpp/pulse/optional.cpp, get_pointer_no_check_none_check_bad, 3, OPTIONAL_EMPTY_ACCESS, no_bucket, ERROR, [in call to `folly::Optional::get_pointer() empty case` (modelled),is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/optional.cpp, FP_value_or_check_value_ok, 5, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `folly::Optional::Optional(Value arg)` (modelled),in call to `folly::Optional::value_or() non-empty case` (modelled),assigned,taking "then" branch,in call to `folly::Optional::Optional(=None)` (modelled),is assigned an empty value,invalid access occurs here]
codetoanalyze/cpp/pulse/optional.cpp, test_trace_ref, 4, OPTIONAL_EMPTY_ACCESS, no_bucket, ERROR, [in call to `folly::Optional<int>::operator=`,in call to `folly::Optional::reset()` (modelled),is assigned an empty value,return from call to `folly::Optional<int>::operator=`,invalid access occurs here]
codetoanalyze/cpp/pulse/path.cpp, only_bad_on_42_latent, 2, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [in call to `may_return_null`,parameter `x` of may_return_null,taking "then" branch,is assigned to the null pointer,returned,return from call to `may_return_null`,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/path.cpp, faulty_call_bad, 0, NULLPTR_DEREFERENCE, no_bucket, ERROR, [calling context starts here,in call to `only_bad_on_42_latent`,null pointer dereference part of the trace starts here,in call to `may_return_null`,parameter `x` of may_return_null,taking "then" branch,is assigned to the null pointer,returned,return from call to `may_return_null`,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/reference_wrapper.cpp, reference_wrapper_heap_bad, 2, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,when calling `getwrapperHeap` here,in call to `WrapsB::WrapsB`,allocated by call to `new` (modelled),assigned,return from call to `WrapsB::WrapsB`,when calling `WrapsB::~WrapsB` here,when calling `WrapsB::__infer_inner_destructor_~WrapsB` here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `getwrapperHeap`,in call to `WrapsB::WrapsB`,allocated by call to `new` (modelled),assigned,return from call to `WrapsB::WrapsB`,in call to `ReferenceWrapperHeap::ReferenceWrapperHeap`,in call to `WrapsB::getb`,returned,return from call to `WrapsB::getb`,assigned,return from call to `ReferenceWrapperHeap::ReferenceWrapperHeap`,return from call to `getwrapperHeap`,invalid access occurs here]
codetoanalyze/cpp/pulse/reference_wrapper.cpp, reference_wrapper_stack_bad, 2, USE_AFTER_LIFETIME, no_bucket, ERROR, [invalidation part of the trace starts here,variable `rw` declared here,when calling `getwrapperStack` here,variable `b` declared here,is the address of a stack variable `b` whose lifetime has ended,use-after-lifetime part of the trace starts here,variable `rw` declared here,in call to `getwrapperStack`,variable `b` declared here,in call to `ReferenceWrapperStack::ReferenceWrapperStack`,assigned,return from call to `ReferenceWrapperStack::ReferenceWrapperStack`,return from call to `getwrapperStack`,invalid access occurs here]
codetoanalyze/cpp/pulse/returns.cpp, returns::return_literal_stack_reference_bad, 0, STACK_VARIABLE_ADDRESS_ESCAPE, no_bucket, ERROR, [variable `C++ temporary` declared here,returned here]
codetoanalyze/cpp/pulse/returns.cpp, returns::return_variable_stack_reference1_bad, 2, STACK_VARIABLE_ADDRESS_ESCAPE, no_bucket, ERROR, [variable `C++ temporary` declared here,assigned,returned here]
codetoanalyze/cpp/pulse/returns.cpp, returns::return_variable_stack_reference2_bad, 3, STACK_VARIABLE_ADDRESS_ESCAPE, no_bucket, ERROR, [variable `C++ temporary` declared here,assigned,assigned,returned here]
codetoanalyze/cpp/pulse/returns.cpp, returns::return_stack_pointer_bad, 2, STACK_VARIABLE_ADDRESS_ESCAPE, no_bucket, ERROR, [variable `x` declared here,returned here]
codetoanalyze/cpp/pulse/shared_ptr_constructors.cpp, shared_ptr_constructors::aliasing_member_null_bad, 4, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `shared_ptr_constructors::ERROR_aliasing_construct_from_internal`,in call to `shared_ptr_constructors::internal_null_def`,is assigned to the null pointer,assigned,return from call to `shared_ptr_constructors::internal_null_def`,in call to `std::shared_ptr<shared_ptr_constructors::A>::shared_ptr<shared_ptr_constructors::B>`,assigned,return from call to `std::shared_ptr<shared_ptr_constructors::A>::shared_ptr<shared_ptr_constructors::B>`,return from call to `shared_ptr_constructors::ERROR_aliasing_construct_from_internal`,in call to `std::shared_ptr<shared_ptr_constructors::A>::operator->`,returned,return from call to `std::shared_ptr<shared_ptr_constructors::A>::operator->`,invalid access occurs here]
codetoanalyze/cpp/pulse/shared_ptr_deref.cpp, shared_ptr::empty_ptr_deref_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::shared_ptr<int>::shared_ptr`,is assigned to the null pointer,assigned,return from call to `std::shared_ptr<int>::shared_ptr`,in call to `std::shared_ptr<int>::operator*`,returned,return from call to `std::shared_ptr<int>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/shared_ptr_deref.cpp, shared_ptr::nullptr_ptr_deref_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::shared_ptr<int>::shared_ptr`,is assigned to the null pointer,assigned,return from call to `std::shared_ptr<int>::shared_ptr`,in call to `std::shared_ptr<int>::operator*`,returned,return from call to `std::shared_ptr<int>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/shared_ptr_deref.cpp, shared_ptr::empty_ptr_field_deref_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::shared_ptr<shared_ptr::X>::shared_ptr`,is assigned to the null pointer,assigned,return from call to `std::shared_ptr<shared_ptr::X>::shared_ptr`,in call to `std::shared_ptr<shared_ptr::X>::get`,returned,return from call to `std::shared_ptr<shared_ptr::X>::get`,invalid access occurs here]
codetoanalyze/cpp/pulse/shared_ptr_deref.cpp, shared_ptr::empty_ptr_field_deref2_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::shared_ptr<shared_ptr::X>::shared_ptr`,is assigned to the null pointer,assigned,return from call to `std::shared_ptr<shared_ptr::X>::shared_ptr`,in call to `std::shared_ptr<shared_ptr::X>::operator->`,returned,return from call to `std::shared_ptr<shared_ptr::X>::operator->`,invalid access occurs here]
codetoanalyze/cpp/pulse/shared_ptr_deref.cpp, shared_ptr::empty_ptr_method_deref_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::shared_ptr<shared_ptr::X>::shared_ptr`,is assigned to the null pointer,assigned,return from call to `std::shared_ptr<shared_ptr::X>::shared_ptr`,in call to `std::shared_ptr<shared_ptr::X>::operator->`,returned,return from call to `std::shared_ptr<shared_ptr::X>::operator->`,invalid access occurs here]
codetoanalyze/cpp/pulse/shared_ptr_deref.cpp, shared_ptr::shared_ptr_copy_null_deref_bad, 2, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/shared_ptr_deref.cpp, shared_ptr::shared_ptr_copy_null_deref_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::shared_ptr<int>::shared_ptr`,is assigned to the null pointer,assigned,return from call to `std::shared_ptr<int>::shared_ptr`,in call to `std::shared_ptr<int>::shared_ptr`,assigned,return from call to `std::shared_ptr<int>::shared_ptr`,in call to `std::shared_ptr<int>::operator*`,returned,return from call to `std::shared_ptr<int>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/std_atomics.cpp, atomic_test::pre_increment_decrement_test_bad, 6, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/std_atomics.cpp, atomic_test::load_store_possible_npe_bad, 5, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/std_atomics.cpp, atomic_test::exchange_possible_npe_bad, 5, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/std_atomics.cpp, atomic_test::FP_compare_exchange_weak_impossible_npe1_ok, 6, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/std_atomics.cpp, atomic_test::compare_exchange_weak_possible_npe1_bad, 6, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/std_atomics.cpp, atomic_test::FP_compare_exchange_weak_impossible_npe2_ok, 6, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/std_atomics.cpp, atomic_test::compare_exchange_weak_possible_npe2_bad, 6, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/std_atomics.cpp, atomic_test::FP_compare_exchange_strong_impossible_npe1_ok, 6, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/std_atomics.cpp, atomic_test::compare_exchange_strong_possible_npe1_bad, 6, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/std_atomics.cpp, atomic_test::FP_compare_exchange_strong_impossible_npe2_ok, 6, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/std_atomics.cpp, atomic_test::compare_exchange_strong_possible_npe2_bad, 6, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/temporaries.cpp, temporaries::call_mk_UniquePtr_A_deref_bad, 4, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,in call to `temporaries::mk_UniquePtr_A`,allocated by call to `new` (modelled),in call to `temporaries::UniquePtr<temporaries::A>::UniquePtr`,assigned,return from call to `temporaries::UniquePtr<temporaries::A>::UniquePtr`,return from call to `temporaries::mk_UniquePtr_A`,when calling `temporaries::UniquePtr<temporaries::A>::~UniquePtr` here,when calling `temporaries::UniquePtr<temporaries::A>::__infer_inner_destructor_~UniquePtr` here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `temporaries::mk_UniquePtr_A`,allocated by call to `new` (modelled),in call to `temporaries::UniquePtr<temporaries::A>::UniquePtr`,assigned,return from call to `temporaries::UniquePtr<temporaries::A>::UniquePtr`,return from call to `temporaries::mk_UniquePtr_A`,in call to `temporaries::UniquePtr<temporaries::A>::get`,returned,return from call to `temporaries::UniquePtr<temporaries::A>::get`,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/temporaries.cpp, temporaries::call_mk_UniquePtr_A_copy_object_ok, 1, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/temporaries.cpp, temporaries::bind_temporary_to_const_bad, 6, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,in call to `temporaries::mk_UniquePtr_A`,allocated by call to `new` (modelled),in call to `temporaries::UniquePtr<temporaries::A>::UniquePtr`,assigned,return from call to `temporaries::UniquePtr<temporaries::A>::UniquePtr`,return from call to `temporaries::mk_UniquePtr_A`,when calling `temporaries::UniquePtr<temporaries::A>::~UniquePtr` here,when calling `temporaries::UniquePtr<temporaries::A>::__infer_inner_destructor_~UniquePtr` here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `temporaries::mk_UniquePtr_A`,allocated by call to `new` (modelled),in call to `temporaries::UniquePtr<temporaries::A>::UniquePtr`,assigned,return from call to `temporaries::UniquePtr<temporaries::A>::UniquePtr`,return from call to `temporaries::mk_UniquePtr_A`,in call to `temporaries::UniquePtr<temporaries::A>::get`,returned,return from call to `temporaries::UniquePtr<temporaries::A>::get`,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/trace.cpp, trace_free_bad, 4, NULLPTR_DEREFERENCE, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,in call to `make_alias`,assigned,return from call to `make_alias`,when calling `do_free` here,assigned,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/trace.cpp, trace_free_bad, 4, USE_AFTER_FREE, no_bucket, ERROR, [invalidation part of the trace starts here,in call to `make_alias`,assigned,return from call to `make_alias`,when calling `do_free` here,assigned,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/traces.cpp, access_destructed_string, 6, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,assigned,in call to `std::basic_string::~basic_string()` (modelled),was invalidated by `delete`,use-after-lifetime part of the trace starts here,assigned,in call to `std::basic_string::data()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/uninit.cpp, Uninit::call_init_by_store_ok, 3, PULSE_UNINITIALIZED_VALUE, no_bucket, ERROR, [struct field address `i` created,read to uninitialized value occurs here]
codetoanalyze/cpp/pulse/uninit.cpp, Uninit::call_get_MyClass_feasible_default_bad, 1, PULSE_UNINITIALIZED_VALUE, no_bucket, ERROR, [struct field address `i` created,read to uninitialized value occurs here]
codetoanalyze/cpp/pulse/unique_ptr_deref.cpp, unique_ptr::empty_ptr_deref_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::unique_ptr<int,std::default_delete<int>>::unique_ptr<true,_void>`,is assigned to the null pointer,assigned,in call to `std::__compressed_pair<int*,std::default_delete<int>>::__compressed_pair<int_*,_std::__default_init_tag>`,in call to `std::forward<int_*>`,returned,return from call to `std::forward<int_*>`,in call to `std::__compressed_pair_elem<int*,0,0>::__compressed_pair_elem<int_*,_void>`,in call to `std::forward<int_*>`,returned,return from call to `std::forward<int_*>`,assigned,return from call to `std::__compressed_pair_elem<int*,0,0>::__compressed_pair_elem<int_*,_void>`,return from call to `std::__compressed_pair<int*,std::default_delete<int>>::__compressed_pair<int_*,_std::__default_init_tag>`,return from call to `std::unique_ptr<int,std::default_delete<int>>::unique_ptr<true,_void>`,in call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,in call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,in call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,returned,return from call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/unique_ptr_deref.cpp, unique_ptr::nullptr_ptr_deref_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::unique_ptr<int,std::default_delete<int>>::unique_ptr<true,_void>`,is assigned to the null pointer,assigned,in call to `std::__compressed_pair<int*,std::default_delete<int>>::__compressed_pair<int_*,_std::__default_init_tag>`,in call to `std::forward<int_*>`,returned,return from call to `std::forward<int_*>`,in call to `std::__compressed_pair_elem<int*,0,0>::__compressed_pair_elem<int_*,_void>`,in call to `std::forward<int_*>`,returned,return from call to `std::forward<int_*>`,assigned,return from call to `std::__compressed_pair_elem<int*,0,0>::__compressed_pair_elem<int_*,_void>`,return from call to `std::__compressed_pair<int*,std::default_delete<int>>::__compressed_pair<int_*,_std::__default_init_tag>`,return from call to `std::unique_ptr<int,std::default_delete<int>>::unique_ptr<true,_void>`,in call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,in call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,in call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,returned,return from call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/unique_ptr_deref.cpp, unique_ptr::empty_ptr_field_deref_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::unique_ptr<unique_ptr::X,std::default_delete<unique_ptr::X>>::unique_ptr<true,_void>`,is assigned to the null pointer,assigned,in call to `std::__compressed_pair<unique_ptr::X*,std::default_delete<unique_ptr::X>>::__compressed_pair<ccfc72f89c918377>`,in call to `std::forward<unique_ptr::X_*>`,returned,return from call to `std::forward<unique_ptr::X_*>`,in call to `std::__compressed_pair_elem<unique_ptr::X*,0,0>::__compressed_pair_elem<unique_ptr::X_*,_void>`,in call to `std::forward<unique_ptr::X_*>`,returned,return from call to `std::forward<unique_ptr::X_*>`,assigned,return from call to `std::__compressed_pair_elem<unique_ptr::X*,0,0>::__compressed_pair_elem<unique_ptr::X_*,_void>`,return from call to `std::__compressed_pair<unique_ptr::X*,std::default_delete<unique_ptr::X>>::__compressed_pair<ccfc72f89c918377>`,return from call to `std::unique_ptr<unique_ptr::X,std::default_delete<unique_ptr::X>>::unique_ptr<true,_void>`,in call to `std::unique_ptr<unique_ptr::X,std::default_delete<unique_ptr::X>>::get`,in call to `std::__compressed_pair<unique_ptr::X*,std::default_delete<unique_ptr::X>>::first`,in call to `std::__compressed_pair_elem<unique_ptr::X*,0,0>::__get`,returned,return from call to `std::__compressed_pair_elem<unique_ptr::X*,0,0>::__get`,returned,return from call to `std::__compressed_pair<unique_ptr::X*,std::default_delete<unique_ptr::X>>::first`,returned,return from call to `std::unique_ptr<unique_ptr::X,std::default_delete<unique_ptr::X>>::get`,invalid access occurs here]
codetoanalyze/cpp/pulse/unique_ptr_deref.cpp, unique_ptr::empty_ptr_field_deref2_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::unique_ptr<unique_ptr::X,std::default_delete<unique_ptr::X>>::unique_ptr<true,_void>`,is assigned to the null pointer,assigned,in call to `std::__compressed_pair<unique_ptr::X*,std::default_delete<unique_ptr::X>>::__compressed_pair<ccfc72f89c918377>`,in call to `std::forward<unique_ptr::X_*>`,returned,return from call to `std::forward<unique_ptr::X_*>`,in call to `std::__compressed_pair_elem<unique_ptr::X*,0,0>::__compressed_pair_elem<unique_ptr::X_*,_void>`,in call to `std::forward<unique_ptr::X_*>`,returned,return from call to `std::forward<unique_ptr::X_*>`,assigned,return from call to `std::__compressed_pair_elem<unique_ptr::X*,0,0>::__compressed_pair_elem<unique_ptr::X_*,_void>`,return from call to `std::__compressed_pair<unique_ptr::X*,std::default_delete<unique_ptr::X>>::__compressed_pair<ccfc72f89c918377>`,return from call to `std::unique_ptr<unique_ptr::X,std::default_delete<unique_ptr::X>>::unique_ptr<true,_void>`,in call to `std::unique_ptr<unique_ptr::X,std::default_delete<unique_ptr::X>>::operator->`,in call to `std::__compressed_pair<unique_ptr::X*,std::default_delete<unique_ptr::X>>::first`,in call to `std::__compressed_pair_elem<unique_ptr::X*,0,0>::__get`,returned,return from call to `std::__compressed_pair_elem<unique_ptr::X*,0,0>::__get`,returned,return from call to `std::__compressed_pair<unique_ptr::X*,std::default_delete<unique_ptr::X>>::first`,returned,return from call to `std::unique_ptr<unique_ptr::X,std::default_delete<unique_ptr::X>>::operator->`,invalid access occurs here]
codetoanalyze/cpp/pulse/unique_ptr_deref.cpp, unique_ptr::empty_ptr_method_deref_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::unique_ptr<unique_ptr::X,std::default_delete<unique_ptr::X>>::unique_ptr<true,_void>`,is assigned to the null pointer,assigned,in call to `std::__compressed_pair<unique_ptr::X*,std::default_delete<unique_ptr::X>>::__compressed_pair<ccfc72f89c918377>`,in call to `std::forward<unique_ptr::X_*>`,returned,return from call to `std::forward<unique_ptr::X_*>`,in call to `std::__compressed_pair_elem<unique_ptr::X*,0,0>::__compressed_pair_elem<unique_ptr::X_*,_void>`,in call to `std::forward<unique_ptr::X_*>`,returned,return from call to `std::forward<unique_ptr::X_*>`,assigned,return from call to `std::__compressed_pair_elem<unique_ptr::X*,0,0>::__compressed_pair_elem<unique_ptr::X_*,_void>`,return from call to `std::__compressed_pair<unique_ptr::X*,std::default_delete<unique_ptr::X>>::__compressed_pair<ccfc72f89c918377>`,return from call to `std::unique_ptr<unique_ptr::X,std::default_delete<unique_ptr::X>>::unique_ptr<true,_void>`,in call to `std::unique_ptr<unique_ptr::X,std::default_delete<unique_ptr::X>>::operator->`,in call to `std::__compressed_pair<unique_ptr::X*,std::default_delete<unique_ptr::X>>::first`,in call to `std::__compressed_pair_elem<unique_ptr::X*,0,0>::__get`,returned,return from call to `std::__compressed_pair_elem<unique_ptr::X*,0,0>::__get`,returned,return from call to `std::__compressed_pair<unique_ptr::X*,std::default_delete<unique_ptr::X>>::first`,returned,return from call to `std::unique_ptr<unique_ptr::X,std::default_delete<unique_ptr::X>>::operator->`,invalid access occurs here]
codetoanalyze/cpp/pulse/unique_ptr_deref.cpp, unique_ptr::unique_ptr_release_bad, 2, MEMORY_LEAK_CPP, no_bucket, ERROR, [allocation part of the trace starts here,when calling `unique_ptr::X::X` here,allocated by `new` here,memory becomes unreachable here]
codetoanalyze/cpp/pulse/unique_ptr_deref.cpp, unique_ptr::reset_ptr_null_deref_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,in call to `std::unique_ptr<int,std::default_delete<int>>::reset`,assigned,return from call to `std::unique_ptr<int,std::default_delete<int>>::reset`,in call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,in call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,in call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,returned,return from call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/unique_ptr_deref.cpp, unique_ptr::reset_ptr_null_deref2_bad, 4, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,in call to `std::unique_ptr<int,std::default_delete<int>>::reset`,assigned,return from call to `std::unique_ptr<int,std::default_delete<int>>::reset`,in call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,in call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,in call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,returned,return from call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/unique_ptr_deref.cpp, unique_ptr::unique_ptr_copy_null_deref_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::unique_ptr<int,std::default_delete<int>>::unique_ptr<true,_void>`,is assigned to the null pointer,assigned,in call to `std::__compressed_pair<int*,std::default_delete<int>>::__compressed_pair<int_*,_std::__default_init_tag>`,in call to `std::forward<int_*>`,returned,return from call to `std::forward<int_*>`,in call to `std::__compressed_pair_elem<int*,0,0>::__compressed_pair_elem<int_*,_void>`,in call to `std::forward<int_*>`,returned,return from call to `std::forward<int_*>`,assigned,return from call to `std::__compressed_pair_elem<int*,0,0>::__compressed_pair_elem<int_*,_void>`,return from call to `std::__compressed_pair<int*,std::default_delete<int>>::__compressed_pair<int_*,_std::__default_init_tag>`,return from call to `std::unique_ptr<int,std::default_delete<int>>::unique_ptr<true,_void>`,in call to `std::unique_ptr<int,std::default_delete<int>>::unique_ptr`,in call to `std::unique_ptr<int,std::default_delete<int>>::release`,in call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,in call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,assigned,returned,return from call to `std::unique_ptr<int,std::default_delete<int>>::release`,assigned,in call to `std::__compressed_pair<int*,std::default_delete<int>>::__compressed_pair<int_*,_std::default_delete<int>_>`,in call to `std::forward<int_*>`,returned,return from call to `std::forward<int_*>`,in call to `std::__compressed_pair_elem<int*,0,0>::__compressed_pair_elem<int_*,_void>`,in call to `std::forward<int_*>`,returned,return from call to `std::forward<int_*>`,assigned,return from call to `std::__compressed_pair_elem<int*,0,0>::__compressed_pair_elem<int_*,_void>`,return from call to `std::__compressed_pair<int*,std::default_delete<int>>::__compressed_pair<int_*,_std::default_delete<int>_>`,return from call to `std::unique_ptr<int,std::default_delete<int>>::unique_ptr`,in call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,in call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,in call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,returned,return from call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/unique_ptr_deref.cpp, unique_ptr::unique_ptr_move_null_deref_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::unique_ptr<int,std::default_delete<int>>::unique_ptr`,in call to `std::unique_ptr<int,std::default_delete<int>>::release`,is assigned to the null pointer,assigned,return from call to `std::unique_ptr<int,std::default_delete<int>>::release`,return from call to `std::unique_ptr<int,std::default_delete<int>>::unique_ptr`,in call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,in call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,in call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,returned,return from call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/unique_ptr_deref.cpp, unique_ptr::no_delete_in_leaky_unique_ptr_bad, 0, MEMORY_LEAK_CPP, no_bucket, ERROR, [allocation part of the trace starts here,when calling `unique_ptr::X::X` here,allocated by `new` here,memory becomes unreachable here]
codetoanalyze/cpp/pulse/unknown_functions.cpp, const_no_init_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/unknown_functions.cpp, call_init_with_pointer_value_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_decl_bad, 1, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, source_mod_param_ok, 1, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_in_branch_bad, 3, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, multiple_copies_bad, 2, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, multiple_copies_bad, 4, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_via_constructor_read_bad, 2, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_vec_bad, 0, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_own_vec_bad, 2, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, loop_via_copy_bad, 1, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_in_both_cases_bad, 3, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_in_both_cases_branch_bad, 3, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_modified_after_abort_ok_FP, 1, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_modified_after_abort_ok_FP, 5, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,variable `vec` declared here,was potentially invalidated by `std::vector::push_back()`,use-after-lifetime part of the trace starts here,variable `vec` declared here,in call to `std::vector::at()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_via_model_bad, 1, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, source_modified_before_lib_destructor_ok, 1, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, check_before_custom_destructor_bad, 0, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_vec_name_contains_copy_ok, 1, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [*** SUPPRESSED ***,copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, source_const_fcn_bad, 1, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, source_dispatch_const_fcn_bad, 1, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, source_dispatch_non_const_fcn_bad_FN, 1, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, SwapVector::swap_ok_FP, 1, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, deref_deleted_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,allocated by call to `new` (modelled),assigned,is assigned to the null pointer,null pointer dereference part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `Simple::Simple` here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, deref_deleted_bad, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `Simple::Simple` here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, reassign_field_of_deleted_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,allocated by call to `new` (modelled),assigned,is assigned to the null pointer,null pointer dereference part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, reassign_field_of_deleted_bad, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, double_delete_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,allocated by call to `new` (modelled),assigned,is assigned to the null pointer,null pointer dereference part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, double_delete_bad, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, delete_in_branch_latent, 5, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,allocated by call to `new` (modelled),assigned,is assigned to the null pointer,null pointer dereference part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, delete_in_branch_latent, 5, USE_AFTER_DELETE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, use_in_branch_latent, 4, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,allocated by call to `new` (modelled),assigned,is assigned to the null pointer,null pointer dereference part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `Simple::Simple` here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, use_in_branch_latent, 4, USE_AFTER_DELETE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `Simple::Simple` here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, delete_in_loop_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,allocated by call to `new` (modelled),assigned,is assigned to the null pointer,null pointer dereference part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, delete_in_loop_bad, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, use_in_loop_bad, 4, NULLPTR_DEREFERENCE, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,allocated by call to `new` (modelled),assigned,is assigned to the null pointer,null pointer dereference part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, use_in_loop_bad, 4, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, delete_allocated_then_error_bad, 1, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, delete_allocated_then_error_bad, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, null_call_delete_allocated_then_error_bad, 1, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,when calling `delete_allocated_then_error_bad` here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::reinit_after_explicit_destructor_bad, 4, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,when calling `use_after_destructor::S::~S` here,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,in call to `use_after_destructor::S::operator=`,assigned,return from call to `use_after_destructor::S::operator=`,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::reinit_after_explicit_destructor2_bad, 5, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,when calling `use_after_destructor::S::~S` here,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,in call to `use_after_destructor::S::operator=`,assigned,return from call to `use_after_destructor::S::operator=`,when calling `use_after_destructor::S::~S` here,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::double_destructor_bad, 5, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,when calling `use_after_destructor::S::~S` here,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,when calling `use_after_destructor::S::~S` here,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::use_after_destructor_bad, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,when calling `use_after_destructor::S::~S` here,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::use_after_scope1_bad, 7, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,when calling `use_after_destructor::S::~S` here,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,in call to `use_after_destructor::S::operator=`,assigned,return from call to `use_after_destructor::S::operator=`,when calling `use_after_destructor::S::~S` here,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::use_after_scope2_bad, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,when calling `use_after_destructor::S::~S` here,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,in call to `use_after_destructor::S::operator=`,assigned,return from call to `use_after_destructor::S::operator=`,when calling `use_after_destructor::S::~S` here,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::placement_new_wrong_order_bad, 4, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,when calling `use_after_destructor::S::~S` here,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,when calling `use_after_destructor::S::~S` here,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::placement_new_aliasing1_bad, 5, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,in call to `<placement new>()` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::placement_new_aliasing2_bad, 5, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,in call to `<placement new>()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::placement_new_aliasing3_bad, 6, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::use_after_scope4_bad, 6, USE_AFTER_LIFETIME, no_bucket, ERROR, [invalidation part of the trace starts here,variable `c` declared here,is the address of a stack variable `c` whose lifetime has ended,use-after-lifetime part of the trace starts here,variable `c` declared here,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, use_after_free_simple_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, use_after_free_simple_bad, 2, USE_AFTER_FREE, no_bucket, ERROR, [invalidation part of the trace starts here,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, double_free_simple_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, double_free_simple_bad, 2, USE_AFTER_FREE, no_bucket, ERROR, [invalidation part of the trace starts here,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, double_free_global_bad, 2, USE_AFTER_FREE, no_bucket, ERROR, [invalidation part of the trace starts here,global variable `global_pointer` accessed here,when calling `free_global_pointer_ok` here,global variable `global_pointer` accessed here,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,global variable `global_pointer` accessed here,when calling `free_global_pointer_ok` here,global variable `global_pointer` accessed here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, free_null_then_deref_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, assumed_aliasing_latent, 3, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, assumed_aliasing_latent, 3, USE_AFTER_FREE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, trigger_assumed_aliasing_bad, 0, NULLPTR_DEREFERENCE, no_bucket, ERROR, [*** SUPPRESSED ***,calling context starts here,in call to `assumed_aliasing_latent`,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, trigger_assumed_aliasing_bad, 0, USE_AFTER_FREE, no_bucket, ERROR, [calling context starts here,in call to `assumed_aliasing_latent`,invalidation part of the trace starts here,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, assumed_aliasing2_latent, 4, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, assumed_aliasing2_latent, 4, USE_AFTER_FREE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, trigger_assumed_aliasing2_bad, 0, NULLPTR_DEREFERENCE, no_bucket, ERROR, [*** SUPPRESSED ***,calling context starts here,in call to `assumed_aliasing2_latent`,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, trigger_assumed_aliasing2_bad, 0, USE_AFTER_FREE, no_bucket, ERROR, [calling context starts here,in call to `assumed_aliasing2_latent`,invalidation part of the trace starts here,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, assumed_aliasing3_latent, 4, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, assumed_aliasing3_latent, 4, USE_AFTER_FREE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, trigger_assumed_aliasing3_bad, 0, NULLPTR_DEREFERENCE, no_bucket, ERROR, [*** SUPPRESSED ***,calling context starts here,in call to `assumed_aliasing3_latent`,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, trigger_assumed_aliasing3_bad, 0, USE_AFTER_FREE, no_bucket, ERROR, [calling context starts here,in call to `assumed_aliasing3_latent`,invalidation part of the trace starts here,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_scope.cpp, access_out_of_scope_stack_ref_bad, 3, USE_AFTER_LIFETIME, no_bucket, ERROR, [invalidation part of the trace starts here,variable `p` declared here,when calling `invalidate_local_ok` here,variable `t` declared here,is the address of a stack variable `t` whose lifetime has ended,use-after-lifetime part of the trace starts here,variable `p` declared here,in call to `invalidate_local_ok`,variable `t` declared here,assigned,return from call to `invalidate_local_ok`,invalid access occurs here]
codetoanalyze/cpp/pulse/values.cpp, error_under_true_conditionals_bad, 5, NULLPTR_DEREFERENCE, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/values.cpp, error_under_true_conditionals_bad, 5, USE_AFTER_FREE, no_bucket, ERROR, [invalidation part of the trace starts here,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/values.cpp, free_if_deref_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,when calling `free_if` here,is assigned to the null pointer,null pointer dereference part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/values.cpp, free_if_deref_bad, 2, USE_AFTER_FREE, no_bucket, ERROR, [invalidation part of the trace starts here,when calling `free_if` here,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, deref_vector_element_after_push_back_bad, 4, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,was potentially invalidated by `std::vector::push_back()`,use-after-lifetime part of the trace starts here,in call to `std::vector::at()` (modelled),assigned,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, deref_vector_pointer_element_after_push_back_bad, 4, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,was potentially invalidated by `std::vector::push_back()`,use-after-lifetime part of the trace starts here,in call to `std::vector::at()` (modelled),assigned,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, deref_local_vector_element_after_push_back_bad, 4, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,variable `C++ temporary` declared here,was potentially invalidated by `std::vector::push_back()`,use-after-lifetime part of the trace starts here,variable `C++ temporary` declared here,in call to `std::vector::at()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, deref_null_local_vector_element_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, FP_init_fill_then_push_back_ok, 7, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,variable `vec` declared here,was potentially invalidated by `std::vector::push_back()`,use-after-lifetime part of the trace starts here,variable `vec` declared here,in call to `std::vector::at()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, push_back_loop_latent, 6, VECTOR_INVALIDATION_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,variable `vec` declared here,was potentially invalidated by `std::vector::push_back()`,use-after-lifetime part of the trace starts here,variable `vec` declared here,in call to `std::vector::at()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, reserve_bad, 3, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,was potentially invalidated by `std::vector::reserve()`,use-after-lifetime part of the trace starts here,in call to `std::vector::at()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, clear_bad, 3, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,was potentially invalidated by `std::vector::clear()`,use-after-lifetime part of the trace starts here,in call to `std::vector::at()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, assign_bad, 3, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,was potentially invalidated by `std::vector::assign()`,use-after-lifetime part of the trace starts here,in call to `std::vector::at()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, shrink_to_fit_bad, 3, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,was potentially invalidated by `std::vector::shrink_to_fit()`,use-after-lifetime part of the trace starts here,in call to `std::vector::at()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, insert_bad, 3, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,was potentially invalidated by `std::vector::insert()`,use-after-lifetime part of the trace starts here,in call to `std::vector::at()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, emplace_bad, 3, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,was potentially invalidated by `std::vector::emplace()`,use-after-lifetime part of the trace starts here,in call to `std::vector::at()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, emplace_back_bad, 3, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,was potentially invalidated by `std::vector::emplace_back()`,use-after-lifetime part of the trace starts here,in call to `std::vector::at()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector_iterator.cpp, iterator_read_after_emplace_bad, 3, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,was potentially invalidated by `std::vector::emplace()`,use-after-lifetime part of the trace starts here,variable `iter` declared here,in call to `std::vector::begin()` (modelled),invalid access occurs here]
codetoanalyze/cpp/pulse/vector_iterator.cpp, iterator_next_after_emplace_bad, 3, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,was potentially invalidated by `std::vector::emplace()`,use-after-lifetime part of the trace starts here,variable `iter` declared here,in call to `std::vector::begin()` (modelled),invalid access occurs here]
codetoanalyze/cpp/pulse/vector_iterator.cpp, iterator_after_push_back_loop_latent, 9, VECTOR_INVALIDATION_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,variable `vec` declared here,was potentially invalidated by `std::vector::push_back()`,use-after-lifetime part of the trace starts here,variable `iter_begin` declared here,in call to `std::vector::begin()` (modelled),in call to `iterator constructor` (modelled),invalid access occurs here]
codetoanalyze/cpp/pulse/vector_iterator.cpp, iterator_end_read_bad, 3, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,is pointed to by the `end()` iterator,use-after-lifetime part of the trace starts here,variable `iter` declared here,in call to `std::vector::end()` (modelled),invalid access occurs here]
codetoanalyze/cpp/pulse/vector_iterator.cpp, iterator_end_next_bad, 3, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,is pointed to by the `end()` iterator,use-after-lifetime part of the trace starts here,variable `iter` declared here,in call to `std::vector::end()` (modelled),invalid access occurs here]
codetoanalyze/cpp/pulse/vector_iterator.cpp, iterator_prev_after_emplace_bad, 4, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,was potentially invalidated by `std::vector::emplace()`,use-after-lifetime part of the trace starts here,variable `iter` declared here,in call to `std::vector::begin()` (modelled),invalid access occurs here]
codetoanalyze/cpp/pulse/weak_ptr.cpp, weak_ptr_observers::use_count_empty_bad, 4, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to function `std::weak_ptr<int>::use_count` with no summary,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/weak_ptr.cpp, weak_ptr_observers::FN_use_count_after_reset_bad, 4, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [in call to function `std::weak_ptr<int>::use_count` with no summary,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/weak_ptr.cpp, weak_ptr_observers::expired_empty_bad, 4, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to function `std::weak_ptr<int>::expired` with no summary,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/weak_ptr.cpp, weak_ptr_observers::FN_expired_after_reset_bad, 4, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [in call to function `std::weak_ptr<int>::expired` with no summary,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/weak_ptr.cpp, weak_ptr_observers::FN_expired_after_swap_bad, 5, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [in call to function `std::weak_ptr<int>::expired` with no summary,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
